### ISSUE TITLE
cant get the code to work? works on psp?

### ISSUE THREAD

**Issue by subject546** [andibadra/HarvestMoon-3DS---XBB-PP-Archive#2]
hi,

I have found some xbb files in a psp game and i wondered if i can decompress those with this program?
however i can't get the code to do anything / complie... i have never worked with c# so maybe i am doing something wrong.
could you provide a step by step guide to compiling / using these files?

subject546

**Comment by andibadra** [comment]
can you give me a sample file for it? 
sometimes is used different structure with using same file extension.

**Comment by subject546** [comment]
the file i am trying to decrypt is on filedropper http://www.filedropper.com/hondaap1race 
there is also a .bin.gz ([honda_ap1_race.bin.gz](https://github.com/andibadra/HarvestMoon-3DS---XBB-PP-Archive/files/495172/honda_ap1_race.bin.gz)) with a bin in it... but i think that might be the same

**Comment by andibadra** [comment]
is seems used same XBB format (and is openend on my old and bm4 internal archive xbb/pp tool) and is seems have duplicate dummy file on it (since their name same, so is have identical filename checksum).

is containing gim, and gmo file right? is containing 333 file, but stripped to 116 file since have duplicate entries for dummy file (0 byte). Here 
[ExtractedFile.zip](https://github.com/andibadra/HarvestMoon-3DS---XBB-PP-Archive/files/496266/ExtractedFile.zip).

my code on this code is only unpacking file from xbb file.

**Comment by subject546** [comment]
OMG! SO HYPED! that is awesome !

tl;dr at bottom

i just realized though i have not given any proper explanation for extracting this archive.
I am trying to learn about my favorite game "Initial D: Street Stage" because it shares close resemblance to the initial d arcade games. I also would like to learn some more about the way psp games are build.

now on to the results! they seem to be exactly what i am looking for! the game is build from a lot of these .bin files but also .gim and .gmo as the texture and 3d-model formats (they can be viewed with a program called noesis) and the extracted files do indeed resemble the s2000 from the game and its textures.

now there was a reason i send you specifically this car (and sorry i did not disclose this to you) but i found a "in the clear" car in the iso of the game, this car was put in a folder called test and might have been a relic of the development life circle of the game (plz correct me if this makes no sense at all ;p). I was hoping this folder would show a resemblance with the extracted archive since the models in that folder started with 'ap1_'.
AND THEY ARE INDEED FROM THE SAME CAR!!!
these models in the test folder however do seem to be composed from the "parts" in the extracted archive and form more functional models (as if a few models from the archive are put together)
ill add them as a zip here: http://www.filedropper.com/test_24

tl;dr ---
i found this car unpacked inside the iso and the extracted models do resemble the car.
---

there is however a strange .bin among your extracted files... it does not seem to have a proper header? or am i wrong? (maybe just a side effect of your code or posably the checksum?)

now back to the question, could you give me instructions on how to use you program so i can extract all the other files from this game as a lot of these models are packed inside xbb archives? (or alternatively extract them for me? if so ill provide a link to the iso and ill check witch files are xbb) 

and thank you so much! this is already a achievement knowing this PSP game shares a strange archive format with a ds3 game and a different developer (since the formats in PSP development where pretty much unregulated there is a huge difference in file types per game).

**Comment by andibadra** [comment]
please don't upload file on filedropper. i can't download from that filehosting >.>

**Comment by subject546** [comment]
this better? https://www.mediafire.com/folder/d9tb8in2y9cmf/test
